platform/commonUI/general/src/ui/TreeNodeView.js

```diff
@@ -150,6 +145,21 @@ define([
         return this.li;
     };
 
+    TreeNodeView.prototype.hasChildren = function (domainObject) {
+
+        if (domainObject) {
+            if (domainObject.hasCapability('composition')) {
+                var proto = Object.getPrototypeOf(domainObject);
```

Comment:
@aayusharora - you can use domainObject.getModel(), which would return the model of the object, to keep consistent with other code.